Share dialog files sidebar

blizzz opened the pull request:
supposed to implement #17665 

obsoletes https://github.com/owncloud/core/pull/18078
- [x] TODO: modularize share dialog into reusable view and item model
- [x] ~~TODO: ensure backwards compatibility with showDropDown()~~ not needed
- [x] TODO: embed share dialog in share tab
- [x] TODO: write unit tests
- [X] REQUIRES: https://github.com/owncloud/core/issues/9227

currently this is based on files-rightsidebar

cc  @PVince81

blizzz committed:
refactor share dialog for multi-purpose use (dropdown, sidebar) and better maintainability

blizzz committed:
share dialog view takes advantage of OC.Backbone.View

blizzz committed:
remove possibly unreliable code and fix JSdoc

blizzz committed:
started to port the Model to Backbone`s

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/ff810acc-cb28-4a55-8d71-3bb91cfe1f9c) was created.

blizzz commented on apps/files_sharing/js/sharetabview.js:
```diff
@@ -10,7 +10,7 @@
 (function() {
 	var TEMPLATE =
-		'<div>Owner: {{owner}}';
+		'<div><span>Owner: {{owner}}</span>';
```
https://github.com/owncloud/core/pull/18078#discussion_r36297548

blizzz commented on core/js/shareitemmodel.js:
```diff
@@ -0,0 +1,104 @@
+
+		_retrieveData: function() {
+			/** var {OC.Share.Types.ShareItemInfo} **/
+			var data = OC.Share.loadItem(this.get('itemType'), this.get('itemSource'));
```
https://github.com/owncloud/core/pull/18078#discussion_r36298030

blizzz replied:
https://github.com/owncloud/core/pull/18078#discussion_r36617290

blizzz commented on core/js/share.js:
```diff
@@ -371,6 +377,26 @@ OC.Share={
 		});
 	},
 	showDropDown:function(itemType, itemSource, appendTo, link, possiblePermissions, filename) {
+		var attributes = {itemType: itemType, itemSource: itemSource};
```
https://github.com/owncloud/core/pull/18078#discussion_r36302282

blizzz replied:
> Here call OC.Share.loadItem() which is still sync ajax, then inject it into the model.

> However in the sidebar it needs to be async, so probably need to rewrite another OC.Share.loadItem

The model needs to take care of interacting with the data, including loading. It only takes the required ID(s) to request it.

DeepDiver1975 commented:
So this code piece continues to use core/ajax/share.php up to now - right?

With moving this to the OCS Sharing API we have an issue as discussed in https://github.com/owncloud/core/pull/17143#issuecomment-115206330

Should we build a files only share side bar within files_sharing and use files_sharing only?
Or move ocs sharing api to core?

DeepDiver1975 commented:
@jancborchardt @PVince81 @blizzz did we respect the share expiry for internal shares in the ui?
https://github.com/owncloud/core/pull/15459#issuecomment-90978180

No need to implement this right away - but we need to respect this in the UI design - THX

PVince81 commented:
>Should we build a files only share side bar within files_sharing and use files_sharing only?

@DeepDiver1975 this is the kind of decision that @schiesbn wanted to have with @rullzer at the conf.

I personally would vote for having a files-only share dialog for now and extend it later to make it generic, not the other way around. But I think @blizzz was also for having a generic sharing mechanism.

DeepDiver1975 commented:
but talking about this at the conf is a bit late? We are building this now.

furthermore - if I remember correctly - we wanted to have share.js untouched - right?

Well - maybe I did miss some parts of the conversation :cry:

blizzz commented:
I like to avoid to maintain two share dialogs and maintain and touch both of them any time we need to fix or extend it. Also, for know I just want to complete this work and have the share dialog integrated in sidebar for files. I prefer new features in a later pull request for not letting this one grow too big and consume too much time.

DeepDiver1975 commented:
> I like to avoid to maintain two share dialogs and maintain and touch both of them any time we need to fix or extend it.

well - from my pov share.js has to simply die.
Any app with the intention of implementing sharing can us core's sharing functionalities but the js side is app specific.

At least this is what we have been talking about @schiesbn @PVince81 @karlitschek 

Don't get me wrong I'm not saying we need to address this now ... but we better keep this in mind

blizzz commented:
it will not receive improvements, but will be thinned out.

DeepDiver1975 commented:
> @DeepDiver1975 this is the kind of decision that @schiesbn wanted to have with @rullzer at the conf.

ok - agreed - please make sure I'm with you when you talk :wink:

blizzz committed:
model now extends Backbone's model and isadjusted

jancborchardt commented:
@blizzz let me know if there’s anything to review interface-wise. The dialog itself doesn’t change in this iteration, it just moves to the sidebar, right?

DeepDiver1975 commented:
> The dialog itself doesn’t change in this iteration, it just moves to the sidebar, right?

it is being reimplemented

blizzz commented:
Which means it stays the same, yes.

blizzz committed:
make loadItems work async if a callback is provided

not beautiful, but good enough. share.js will not be around too much longer.

blizzz committed:
switch to async item loading, take care stuff is updated subsequently

blizzz committed:
improve reshare rendering part and move permission calculation to model

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/3c4f8bfd-4bba-40bd-b5f9-8c0c989c1858) was created.

blizzz committed:
further work on reimplementing the dialog layout

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/019dfc8f-10a7-4761-832d-4289a37ceba3) was created.

PVince81 commented on core/js/share.js:
```diff
@@ -289,21 +289,32 @@ OC.Share={
+	 *
+	 * @param itemType
+	 * @param itemSource
+	 * @param callback
```
Make sure to document the callback vs no callback sync behavior

PVince81 commented on core/js/shareconfigmodel.js:
```diff
@@ -0,0 +1,54 @@
+		OC.Share.Types = {};
+	}
+
+	var ShareConfigModel = OC.Backbone.Model.extend({
```
You're only half-using a model.

To use it properly you should ".set()" the properties on the model.
Then consumers can call model.get('publicUploadEnabled') to get the config value.

You could add the DOM-parsing code in `initialize()`. Normally `initialize` would pass you the config values.

PVince81 replied:
or get rid of Backbone Model and leave this as a simple static JS class.

PVince81 commented on core/js/sharedialogview.js:
```diff
@@ -0,0 +1,320 @@
+				expiration: this._renderExpirationPart()
+			}));
+
+			return this;
```
Pro-tip: at some point you'll probably use the "events: {}" attribute for jquery events.
If you do, you'll need to add `this.delegateEvents()` at the end of the render method.
Got bitten by this yesterday. :smile:

PVince81 commented on core/js/sharedialogview.js:
```diff
@@ -0,0 +1,320 @@
+		'<a target="_blank" class="icon-info svg shareWithRemoteInfo" href="{{docLink}}" ' +
+		'title="{{tooltip}}"></a>';
+
+	var TEMPLATE_LINK_SHARE =
```
I wonder if we should split the different pieces into separate Views so we can also have the logic for the different parts neatly organized in separate class.

As you like, I'm fine with keeping this in a single view for now.

blizzz replied:
It is a valid question, that was around in my head as well. It's certainly better split and structured right now to the mess before. But the longer and more crowded the view gets, the uneasier i feel about this too. OTOH, having every single piece in different views, well, that might be just overkill. Although, when it is about updating, things will get easier as views could just listen to what they need instead of the redo-everything approach right now.

blizzz replied:
As discussed and realized later, this split up in separate views is preferred.

PVince81 commented on core/js/sharedialogview.js:
```diff
@@ -0,0 +1,320 @@
+
+		_renderSharePlaceholderPart: function () {
+			var sharePlaceholder = t('core', 'Share with users or groups …');
+			if (oc_appconfig.core.remoteShareAllowed) {
```
use this.configModel here :smile:

PVince81 commented on core/js/shareitemmodel.js:
```diff
@@ -0,0 +1,200 @@
+		fetch: function() {
+			var model = this;
+			OC.Share.loadItem(this.get('itemType'), this.get('itemSource'), function(data) {
+				model.set(model.parse(data));
```
Not sure if this is the right way.
Normally we shouldn't override fetch but rather provide the endpoint url as attribute in the class.
But since we're using the old API to do it, then it's fine to override fetch() for that.
What is missing is calling the success callback.

You can leave it as is now, I'll see how I can improve this later.

blizzz replied:
This is until we have an API that we can use. This way it's easier than to play around with URLs and stuff for now, especially since it should not stay like this for long. 

The callback is called in any case, covering Success and Error. Actually this stuff is dealt with priorly in loadItems and either real data or empty something is delivered.

jancborchardt commented:
This should also be rebased to reflect the actions in the dropdown change in master.

PVince81 commented:
also note: the DetailsView and subviews are now backbone views.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/f1c5d3df-9a4a-43d7-88ec-f3e3c489c1b3) was created.

blizzz commented:
*rebased*

blizzz committed:
move remaining global settings to configModel

blizzz committed:
improve doc

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/c29ef7bf-701c-4101-b228-552d6a2ef80f) was created.

blizzz committed:
show tooltips and load avatar

blizzz committed:
split ShareDialogResharerInfoView from base view

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/3719a974-2045-4e3c-aa71-10c69e0a05ed) was created.

PVince81 commented on core/js/sharedialogresharerinfoview.js:
```diff
@@ -0,0 +1,117 @@
+			var view = this;
+
+			//FIXME: specific to reshares stuff
+			this.model.on('change', function() {
```
with a bit of luck "change:reshares" could work :wink:

blizzz replied:
yes, should be safe, thanks for the hint

PVince81 commented on core/js/sharedialogview.js:
```diff
@@ -0,0 +1,300 @@
+
+			this.$el.html(baseTemplate({
+				shareLabel: t('core', 'Share'),
+				resharerInfo: this.resharerInfoView.el.innerHTML,
```
I suggest to not inject HTML from outside like this.
Just provide a container in this view and then do `this.$el.find('.somecontainer').append(this.resharerInfoView.el)`

PVince81 replied:
This is necessary because you need the real element to be in the container, not a copy of its HTML.
You'll need it when re-rendering sub-views.

blizzz replied:
That's a good point. I had the idea with the div running in my head already, but didn't follow. Will go that approach.

PVince81 commented on core/js/sharedialogview.js:
```diff
@@ -0,0 +1,300 @@
+	}
+
+	var TEMPLATE_BASE =
+		'{{{resharerInfo}}}' +
```
provide an empty div container instead. Then append the subview's $el element in the render method.

PVince81 commented on core/js/shareitemmodel.js:
```diff
@@ -0,0 +1,201 @@
+	 * @property {string} share_with
+	 * @property {string} displayname_owner
+	 * @property {number} permissions
+	 */
```
Nice stuff!

PVince81 commented on core/js/shareitemmodel.js:
```diff
@@ -0,0 +1,201 @@
+		 * @returns {boolean}
+		 */
+		hasReshare: function() {
+			return _.isObject(this.get('reshare')) && !_.isUndefined(this.get('reshare').uid_owner);
```
you can also do `this.has('reshare')`, but not sure if this achieves what you want here

blizzz replied:
It would only replace the first call of this.get(), but better keeping that in local var and call get() just  once instead of twice as now.

blizzz committed:
ShareDialogResharerInfoView improvements

blizzz committed:
split off linkShareView

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/273db25b-a439-4cfa-a588-b80c40f57fb8) was created.

blizzz committed:
split off expirationView

blizzz committed:
less stupid initalization of subviews

blizzz committed:
integraton noshare part into ShareDialogLinkShareView

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/b04dbb1d-fec0-4a54-b997-f5d67662aadd) was created.

PVince81 commented:
- [ ] fix resource order loading, use this approach: https://github.com/owncloud/activity/blob/master/lib/fileshooksstatic.php#L43

Commit:
Add share dialog into share tab

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/292a04e4-0599-4f11-a152-9c29e47adf90) was created.

PVince81 commented:
@blizzz added the dialog into the sidebar: https://github.com/owncloud/core/commit/00f9f9c54d658474863b78f7e4dca9a1605a7fe0

We might need to extract some of the dialog styles from "shareDropdown" to make them reusable in the sidebar.

PVince81 commented:
- [ ] BUG: fix "possiblepermissions" for share tab
- [ ] TODO: loading spinner in share tab, properly update contents when switching files

jancborchardt commented:
Wanted to check out the branch, get a whitescreen at the moment with this in the owncloud.log:
```
Declaration of OCA\\Activity\\Consumer::receive() must be compatible with OCP\\Activity\\IConsumer::receive($app, $subject, $subjectParams, $message, $messageParams, $file, $link, $affectedUser, $type, $priority)
at \/apps\/activity\/lib\/consumer.php#30
```
cc @nickvergessen

PVince81 commented:
- [ ] TODO: rebase this onto master (to make it work properly with the activity tab)
(@jancborchardt please disable the activity app for now, also the share dialog doesn't seem to work yet, so not much to see)

Commit:
Fix share permissions for share tab

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/dc504b8e-b160-41ab-9278-c69e2d9fade9) was created.

PVince81 commented on core/js/shareitemmodel.js:
```diff
@@ -0,0 +1,202 @@
+			if(!_.isUndefined(options.configModel)) {
+				this.configModel = options.configModel;
+			}
+			this.fetch();
```
Don't auto-fetch the model.
The caller is responsible for deciding when to fetch, especially that there is no callback possibility at this stage.

PVince81 commented on core/js/sharedialogview.js:
```diff
@@ -0,0 +1,191 @@
+			if(!_.isUndefined(options.configModel)) {
+				this.configModel = options.configModel;
+			} else {
+				console.warn('missing OC.Share.ShareConfigModel');
```
throw ? Without config it seems impossible to use this class anyway

blizzz replied:
Exactly about this I was thinking. Thanks for the hint.

PVince81 commented on core/js/sharedialoglinkshareview.js:
```diff
@@ -0,0 +1,151 @@
+				|| !this.configModel.isShareWithLinkAllowed())
+			{
+				this.$el.empty();
+				this.$el.append(linkShareTemplate({
```
You can simply use `$this.$el.html(linkShareTemplate(...))` (the backbone examples also do so, see http://backbonejs.org/#View-render)

blizzz commented:
*rebased*

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/676ece6e-8b61-4fe6-a245-069eee0d2b13) was created.

blizzz committed:
simplification, and throwing where throwing is needed

blizzz committed:
no auto-fetch in model

blizzz committed:
set default value and remove now superflous method

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/c33d0efc-799d-4e27-a755-f6fe107a180e) was created.

blizzz committed:
cleanup

blizzz committed:
started to implement sharee list view. not completed yet, do not cry please.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/7a235b36-8104-4aab-b814-a167c5510bff) was created.

blizzz committed:
continue to reimplement sharee list view. still WIP

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/a4be9794-ac49-4c0e-9fe9-032943ee4676) was created.

blizzz committed:
sharee list view: better handle collections

blizzz committed:
old OC.Share.addShareWith now reimplemented

blizzz committed:
fixed set of possible permissions for remote shares

jancborchardt commented:
@Blizzz @rullzer @PVince81 what’s the state here? This is quite key for 8.2

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/ec8b3c28-b057-4303-95d6-53c2db61d50c) was created.

blizzz committed:
show link share

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/08bb67a0-b415-4b79-9990-c73e3b46b271) was created.

blizzz committed:
display expiration info

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/b6165b40-caef-4a65-8152-4cc5bec0d22e) was created.

karlitschek commented:
@blizzz Where are we with that? We need it for 8.2 Thanks

blizzz committed:
now you even can share

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/8e89311d-dc24-4490-bb39-8a8e6a6a0172) was created.

blizzz commented:
@karlitschek 

What is done:
* Displaying is totally unbundled from manipulation. As soon as anything gets changed (and the model pulls it), the current state is shown. This should be feature complete with the share dialogue. While may sound stupid, >1,5k of the new code went inside there, plus the mind work.
* Autocomplete works, sharing with users (includes remote users) is possible. 

What is left:
* Otherwise manipulating shares (as in changing permissions, deleting shares) needs to be implemented. Fortunately, these require just an server call (with little logic necessary, basically copying the call stuff without GUI-handling which will already be taken care off) and request the model to reload (one-liner). This is not too hard or much to do, especially compared to the software archaeology that was required to come to this point.
* Style tuning – the sidebar is smaller then the old dialog. @jancborchardt may want to have a look.
* remove old logic (just some code deletions)

karlitschek commented:
Thanks. @schiesbn @nickvergessen Can you help finishing this and merge it? Thanks

jancborchardt commented:
@blizzz @schiesbn @nickvergessen clicking the Share entry in the row should bring up the sidebar with the Share tab open. Also, can you rebase the branch?

blizzz committed:
remove obsolete file action

blizzz committed:
format groups and remotes in autocomplete list

blizzz committed:
share and unshare via link (not yet password). also some internal changes to reduce unnecessary rendering

blizzz committed:
mark url on click

blizzz committed:
implements setting and removing password for link shares, including forced ones

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/b362a083-967e-4865-b101-e2ad4edf5ec2) was created.

blizzz commented:
Rebased. Link sharing things all work now, including all stuff about password protection. Maybe I be able to add some work later or tomorrow… certainly not Mon/Tue. Cheers from ![Kaliningrad](http://eastbook.eu/wp-content/uploads/2013/06/kaliningrad.jpg)

karlitschek commented:
Thanks a lot @blizzz 
Who has the time to finish this work, review and merge it?

blizzz committed:
fix allow upload checkbox appeareance

blizzz committed:
make 'Allow editing' (aka public upload) checkbox work

blizzz committed:
make unshare work

blizzz committed:
cleanup

blizzz committed:
toggle visibility of detailed cruds permissions

blizzz committed:
make permission/cruds checkboxes work

blizzz committed:
cleanup

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/9087d1e0-f782-46bc-8767-830420160be2) was created.

blizzz commented:
So tonight I made all the cruds buttons, public upload, unshare buttons work and cleaned up some old unnecessary code.

What is left:

* [ ] implement set expiration date (remove showExpirationDate from core/js/share.js afterwards)
* [ ] implement email private link dropdown (remove related event handlers from core/js/share.js afterwards. https://github.com/owncloud/core/blob/master/core/js/share.js#L601 to L621 might be of interest, that's why i left the showDropDown method, although is does not do anything anymore. Remove afterwards to).
* [ ] implement send mail notification

Two things I saw today, but might already be out of scope and bug fixes.

* [ ] cannot share with default expiration date and enforced expiration. Does it work in master?
* [ ] OC.Share.updateIcons does not work (anymore? needs adjustement?)

blizzz commented:
>  Who has the time to finish this work, review and merge it?

Would be great if someone would pick it up, I won't be able anymore now.

PVince81 commented:
Ok, I'll take over this.

PVince81 commented:
- [ ] BUG: spinner of checkboxes breaks layout

karlitschek commented:
Thanks a lot @PVince81

schiesbn commented:
> Two things I saw today, but might already be out of scope and bug fixes.

> * [ ] cannot share with default expiration date and enforced expiration. Does it work in master?

Just checked it... It works on master

PVince81 commented:
- [ ] TODO: bring back the "Share" icon action, it should stay there to show the owner. Clicking it should open the sidebar and focus on the "Share" tab

Commit:
Add expiration date handling in share tab

PVince81 commented:
- [ ] TODO: auto-update share icon info when info modified in the share section (needed for some of the share overviews)

PVince81 commented:
- [ ] BUG: removing user share reloads the whole page

PVince81 commented:
- [ ] BUG: autocomplete box doesn't clear after adding a user

Commit:
Bring back the share icon and update its status

Display share icon in file list row.
Update share icon status when the sharing state changed.

Commit:
Fix shares list events

Some events need preventing default like unshare, but setting
permissions doesn't.

Commit:
Clear field after adding user

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/fb7c50a1-64ee-4a3b-a57d-c81416c37a5f) was created.

PVince81 commented:
> cannot share with default expiration date and enforced expiration. Does it work in master?

Works for me too on this branch.

Commit:
Implement "notify by email" checkbox in share dialog

Commit:
Implement email share link

Commit:
Implement email autocomplete for link share

PVince81 commented:
- [ ] TODO: double check if _parseTime was ported properly, seems to be a different impl

Commit:
Remove unused/obsoleted OC.Share code

The sharing dialog has been moved to ShareDialogView

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/f81690d3-0905-4cec-96ae-65ea51101ed9) was created.

PVince81 commented:
Ok, the features are all there.
This will need some manual regression testing.

Tomorrow I'll port the JS unit tests.

PVince81 commented:
Would be cool if people with some time could help regression testing the functionality for the many combinations / code paths for the share dialog (see admin page checkboxes, etc).
@schiesbn @MorrisJobke @nickvergessen @icewind1991 @DeepDiver1975 

Thanks.

PVince81 commented:
well and review the code too, but keep in mind that many chunks were copy-pasted from the old dialog so you might still glance at some legacy horrors. :wink:

Commit:
Update JS unit tests for share dialog (WIP)

nickvergessen commented:
The info about remote shares is hidden partly because it's outside the box:
![bildschirmfoto vom 2015-09-15 10 41 49](https://cloud.githubusercontent.com/assets/213943/9872033/f3be7982-5b96-11e5-9876-c48008cd8b0c.png)

Also the `(i)` should be on the same line as the box, and the avatar on the same line as user name?
But maybe this can be done in a second PR.

nickvergessen commented:
Enabling "Link share" has a small glitch (the box moves to the left until it's half hidden and then jumps back) as per @fossxplorer
I experience this as well

nickvergessen commented:
Played around a bit with it and couldn't see functionality failures

PVince81 commented:
@nickvergessen yes, the layout as some issues. I'll have a look after I'm done with the JS unit tests, if time permits.

It would also be good to do some reshare testing:
- reshare
- trying resharing with disabled reshare permissions globally
- try resharing when share permissions were not given
- try resharing a child of a share folder (link and regular share)
- ...

PVince81 commented:
Before the freeze I'm mostly aiming at having tests pass and finish the JS unit tests.
If there is no major functionality issues, I suggest to merge and fix the small bugs during the bugfix phase.

icewind1991 commented:
Code looks good at a glance, basic functionality seems to work, various minor interface issues

karlitschek commented:
looks good. Tested this with @PVince81 :+1: 
Design needs some love. @jancborchardt Can you help please?

jancborchardt commented:
Clicking the share action in the files rows does not bring up the sidebar with the sharing tab open.

jancborchardt commented:
And clicking on the »Can edit« further CRUDS options dropdown opens that dropdown for all sharees.

jancborchardt commented:
I’d also say we should merge this for the feature freeze and fix all the issues after that before it grows bigger …

Commit:
More unit tests for share dialog

PVince81 commented:
>Clicking the share action in the files rows does not bring up the sidebar with the sharing tab open.

Please pull.

9 JS unit tests to go...

Commit:
Fix allow reshare for owner when sharing with self through group

Commit:
Removed obsolete tests

Commit:
Fix cruds button in share dialog

Only toggle the current row

PVince81 commented:
I discovered that share.css had some stray styles, will move them to the proper selectors. This will fix a few issues. Will push soon.

Commit:
Port share.css styles to the sidebar panel

PVince81 commented:
I'm done now.

JS unit tests should pass.

The styles are fixed too, it looks better now.
There is still a glitch with the spinner when clicking on checkboxes => to do separately.

Since there were no server-side changes, only JS CI should need to pass (in case Oracle is angry again)

icewind1991 commented:
There still seems to be some lingering old logic

![](https://i.imgur.com/udHnZtM.png)

What is the plan with the "old" share button in the file list?

icewind1991 commented:
Design issues mentioned earlier are gone

jancborchardt commented:
> What is the plan with the "old" share button in the file list?

What do you mean by that? It stays of course as a status indicator. Also when clicked it invokes the sidebar with the sharing tab readily open.

icewind1991 commented:
It seems like needless clutter to me to have it shown for files that aren't shared

jancborchardt commented:
> It seems like needless clutter to me to have it shown for files that aren't shared

It is to enable quick & easy sharing of those files of course. Sharing is the most important of the actions, that is why it’s not tucked away in the menu. Please not detail discussions now on the day of the feature freeze about things we defined before and decided on with good reason.

icewind1991 commented:
> Please not detail discussions now

Sure, was just wondering what the plan was atm

PVince81 commented:
Rebased locally, and the new "Details" menu item logic causes trouble:
- [ ] BUG: click share action doesn't work after rebasning

Commit:
Properly show sidebar when clicking on share button

PVince81 commented:
Fixed in this one-liner: https://github.com/owncloud/core/commit/e64799f08fca6a1b549db3f0c5a146f08c9fd84d

icewind1991 commented:
All good :+1:

PVince81 commented:
Now let's hope the Oracle is in a good mood today.
Github won't let us merge this even if it's JS only.

rullzer commented:
As mentioned on IRC already. Code looks good. And most functionality seems to work as expected.
I also vote to merge this and for 8.2 and fix the remaining issues in the next few days. :+1: from me.

Awesome work @blizzz and @PVince81!

PVince81 commented:
@owncloud-bot retest this please

DeepDiver1975 commented:
> @owncloud-bot retest this please

@PVince81 will no longer help - you need to rebase and force push

PVince81 commented:
Force-pushed. Let's hope CI will be kind this time.

DeepDiver1975 commented:
jsunit and phpunit on sqlite did succeed -> merge

jancborchardt commented:
Wow. WOW!

Finally. :D great work @blizzz @PVince81

jospoortvliet commented:
Congratulations everybody ;-)

This is awesome stuff.
